pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_controller.go

```diff
@@ -77,23 +83,24 @@ func NewAPIReconciler(
 		queue: queue,
 
 		createAPIDefinition: createAPIDefinition,
+		allowedAPIfilter:    allowedAPIfilter,
 
 		apiSets: map[dynamiccontext.APIDomainKey]apidefinition.APIDefinitionSet{},
 	}
 
 	if err := syncTargetInformer.Informer().AddIndexers(cache.Indexers{
-		indexSyncTargetsByExport: indexSyncTargetsByExports,
+		indexSyncTargetsByExport + virtualWorkspaceName: indexSyncTargetsByExports,
```

Comment:
I'm still worried about duplicating the indexes here. Can't they be setup before entering this method, and be common to both controllers ?

Reply:
done!